From fb48ab3594972dc7304b66604b4747e1c9aa4926 Mon Sep 17 00:00:00 2001
From: Alex Harries <alex@greyhead.co.uk>
Date: Tue, 23 Feb 2016 18:35:10 +0000
Subject: [PATCH] Geocoder: geocoder_widget_parse_addressfield - allow the
 handling of addresses where the address array contains only a
 pre-concatenated value array key.

Signed-off-by: Alex Harries <alex@greyhead.co.uk>
---
 geocoder.widget.inc | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/geocoder.widget.inc b/geocoder.widget.inc
index a50cec6..7c7ce5d 100644
--- a/geocoder.widget.inc
+++ b/geocoder.widget.inc
@@ -517,7 +517,13 @@ function geocoder_widget_parse_addressfield($field_item) {

   $address[] = !empty($field_item['postal_code']) ? $field_item['postal_code'] : NULL;

-  return implode(',', array_filter($address));
+  $return = implode(',', array_filter($address));
+
+  if (empty($return) && array_key_exists('value', $field_item)) {
+    $return = $field_item['value'];
+  }
+
+  return $return;
 }

 /**
--
2.5.4 (Apple Git-61)

